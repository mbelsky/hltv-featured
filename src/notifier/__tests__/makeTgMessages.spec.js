const { makeTgMessages } = require('../makeTgMessages')

const rawMessages = [
  {
    id: -1,
    featuredMatches: { message: 'featuredMatches', type: 'EMPTY' },
  },
  {
    id: 0,
    featuredMatches: { message: 'featuredMatches', type: 'SUCCESS' },
  },
  {
    id: 1,
    favoriteTeamsMatches: 'favoriteTeamsMatches',
  },
  {
    id: 2,
    customLocationMessage: 'customLocationMessage',
  },
  {
    id: 3,
  },
]

describe('makeTgMessages', () => {
  test('common', () => {
    expect(makeTgMessages(rawMessages)).toMatchSnapshot()
  })
})
