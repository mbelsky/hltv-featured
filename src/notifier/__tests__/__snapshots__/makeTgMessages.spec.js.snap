// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeTgMessages common 1`] = `
Array [
  Object {
    "id": -1,
    "messages": Array [
      Object {
        "extra": Object {
          "disable_notification": true,
          "disable_web_page_preview": true,
          "parse_mode": "HTML",
        },
        "text": "featuredMatches",
        "type": "emptyFeaturedMatches",
      },
    ],
  },
  Object {
    "id": 0,
    "messages": Array [
      Object {
        "extra": Object {
          "disable_notification": false,
          "disable_web_page_preview": true,
          "parse_mode": "HTML",
        },
        "text": "featuredMatches",
        "type": "featuredMatches",
      },
    ],
  },
  Object {
    "id": 1,
    "messages": Array [
      Object {
        "extra": Object {
          "disable_web_page_preview": true,
          "parse_mode": "HTML",
        },
        "text": "favoriteTeamsMatches",
        "type": "favoriteTeamsMatches",
      },
    ],
  },
  Object {
    "id": 2,
    "messages": Array [
      Object {
        "text": "customLocationMessage",
        "type": "customLocation",
      },
    ],
  },
]
`;
